### DISCUSSION TITLE
DApp design

### DISCUSSION THREAD

**Discussion by patrickdronk** [spartan-protocol/SpartanProtocol-DAppV2#412] [General]
Hey all. 

Recently I've started contributing to the DAppV2 of spartan protocol. 
I've been working on the latest transactions feature, and before that some simple issues. 
There is one thing that, I think, could have more impact on the future, assuming growth of the protocol and its contributors. 
That thing is consistency and a guideline into developing things. 

I've noticed that we are using several component libraries (react-bootstrap, reactstrap and some other bootstrap related things), and writing quite a lot ourselves as well. Which itself is a good thing. But reinventing the wheel  doesn't go without making some, if not a lot, mistakes again. (there are quite some bugs on the DApp related to theming and sidebar, almost all visually, but stil)

Also, even though I'm not the most experienced react developer out there (I've had my fair share of time with several projects. ) I would also like to consider the need redux in its current form. I've noticed that with the introduction of react hooks and react context, the necessity of redux has reduced. And I believe hooks and context are easier to use.

Summarizing, I would like to know your thoughts:

- Should we use one component library and stick with the features provided by it (I'm personally quite fond of Antd). 
- Should we have a contributing guidelines and how should it look.

Hoping my written thoughts makes any sense to you all.

**Comment by SamusElderg** [comment]
Hey buddy! Thanks for the feedback, totally agree on most fronts (as we discussed on Telegram)
Maybe we should make 3 github issues following this for:
- Cull/cut back component libraries to just 'reactstrap' or similar, pick and choose with some scrutiny to add any others after that (charting lib? etc) been meaning to do this for ages; dappv2 was always slated to be minimalist but unfortunately it got bloated over time; I only use ReactStrap personally, not sure where the rest of it is injected or if it even is; ill have to do some digging 👍 
- Contributing guidelines 
- Replace the sidebar/nav/header with something a bit less patched together but still follow the Figma styling as a guide (will result in pretty much the same style/look but better behaviour)

In regard to Context vs Redux I'm open to assessing whether it will be viable; but however I last used Context wasn't suitable for the size of the dapp state, I have no major experience though so will need to be guided in regard to this :) need to leverage everyone's experience. I was scared of Redux but am now very happy we used it despite its code-bloat. Being able to open up the Redux tools and see every variable/state sectioned into contracts is really awesome and the hard work with Redux is starting to pay off
